domainservice/src/test/kotlin/ua/marchenko/artauction/auction/controller/nats/GetAllAuctionsNatsControllerTest.kt

```diff
@@ -0,0 +1,68 @@
+package ua.marchenko.artauction.auction.controller.nats
+
+import artwork.random
+import auction.random
+import java.time.Clock
+import java.time.LocalDateTime
+import java.time.temporal.ChronoUnit
+import kotlin.test.assertTrue
+import org.junit.jupiter.api.Test
+import org.springframework.beans.factory.annotation.Autowired
+import ua.marchenko.artauction.artwork.mapper.toArtworkProto
+import ua.marchenko.artauction.artwork.model.MongoArtwork
+import ua.marchenko.artauction.artwork.repository.ArtworkRepository
+import ua.marchenko.artauction.auction.mapper.toAuctionProto
+import ua.marchenko.artauction.auction.model.MongoAuction
+import ua.marchenko.artauction.auction.repository.AuctionRepository
+import ua.marchenko.artauction.common.AbstractBaseNatsControllerTest
+import ua.marchenko.internal.NatsSubject
+import ua.marchenko.internal.input.reqreply.auction.FindAllAuctionsRequest as FindAllAuctionsRequestProto
+import ua.marchenko.internal.input.reqreply.auction.FindAllAuctionsResponse as FindAllAuctionsResponseProto
+
+class GetAllAuctionsNatsControllerTest : AbstractBaseNatsControllerTest() {
+
+    @Autowired
+    private lateinit var artworkRepository: ArtworkRepository
+
+    @Autowired
+    private lateinit var auctionRepository: AuctionRepository
+
+    @Autowired
+    private lateinit var clock: Clock
+
+    @Test
+    fun `should return all auctions when they are exists`() {
+        // GIVEN
+        val artworks = List(2) { artworkRepository.save(MongoArtwork.random(id = null)).block()!!.toArtworkProto() }
+        val auctions = listOf(
+            auctionRepository.save(
+                MongoAuction.random(
+                    id = null,
+                    artworkId = artworks[0].id!!,
+                    startedAt = LocalDateTime.now().truncatedTo(ChronoUnit.MILLIS),
+                    finishedAt = LocalDateTime.now().plusDays(1).truncatedTo(ChronoUnit.MILLIS)
+                )
+            ).block()!!.toAuctionProto(clock),
+            auctionRepository.save(
+                MongoAuction.random(
+                    id = null,
+                    artworkId = artworks[1].id!!,
+                    startedAt = LocalDateTime.now().truncatedTo(ChronoUnit.MILLIS),
+                    finishedAt = LocalDateTime.now().plusDays(1).truncatedTo(ChronoUnit.MILLIS)
+                )
+            ).block()!!.toAuctionProto(clock),
+        )
```

Comment:
If we will want to add more artworks to our list to make it, for example, 5, we will just copy this?

Reply:
there are different artworkId, so I cant just write `List (2) {auctionRepository.save(MongoAuction.random...)}`